packages/shared/src/lib/core/layer-2/utils/getTransferInfoFromTransactionData.ts

```diff
@@ -26,7 +27,7 @@ export function getTransferInfoFromTransactionData(
         const abi = isErc20 ? ERC20_ABI : isIscContract ? ISC_SANDBOX_ABI : undefined
 
         if (!abi) {
-            return undefined
+            return { type: 'SmartContract' }
```

Comment:
The smart contract type needs to be returned in the switch statement as well. This solves the issue that `Enable token spend` on Virtue doesn't show up as a smart contract activity.